lib/scala/kg/src/main/scala/formats/CsvReader.scala

```diff
@@ -0,0 +1,51 @@
+package formats
+
+import com.github.tototoshi.csv.CSVReader
+import java.nio.file.Path
+import java.io.BufferedInputStream
+import java.io.FileInputStream
+import java.io.InputStream
+import java.io.InputStreamReader
+import org.apache.commons.compress.compressors.CompressorStreamFactory
+import org.apache.commons.compress.compressors.CompressorException
+import java.io.Reader
+import com.github.tototoshi.csv.CSVFormat
+import org.apache.commons.lang3.StringUtils
+
+abstract class CsvReader[T](protected val csvReader: CSVReader) extends AutoCloseable with Iterable[T] {
+  override def close(): Unit =
+    csvReader.close()
+
+  protected implicit class RowWrapper(row: Map[String, String]) {
+    def getNonBlank(key: String) =
+      row.get(key).flatMap(value => if (!StringUtils.isBlank(value)) Some(value) else None)
```

Comment:
I use this pattern a lot, and only recently discovered that you can do Option.filter, as I mentioned above. So this becomes:
`row.get(key).filter(!StringUtils.isBlank(_))`

That's a great thing about Scala, always new ways to learn.

Unrelated, see the Option .orElse method. I saw Owen Xie using that on DsaModels.

Reply:
Nice changed